src/main/java/spoon/experimental/modelobs/EmptyModelChangeListener.java

```diff
@@ -111,10 +126,24 @@ public void onSetDelete(CtElement currentElement, CtRole role, Set field, Collec
 	@Override
 	public void onSetDelete(CtElement currentElement, CtRole role, Set field,
 			ModifierKind oldValue) {
+		this.changeCu(currentElement);
 	}
 
 	@Override
 	public void onSetDeleteAll(CtElement currentElement, CtRole role, Set field,
 			Set oldValue) {
+		this.changeCu(currentElement);
+	}
+
+	private void changeCu(CtElement element) {
+		SourcePosition position = element.getPosition();
+		if (position != null && !(position instanceof NoSourcePosition)) {
+			position.getCompilationUnit().setIsChanged(true);
```

Comment:
If spoon needs that call to keep internal consistency, then it should be part of some listener, which is always used - cannot be replaced by a different custom listener. I guess spoon should allow to register more then one listener.
WDYT?

Reply:
Not necessarily.
It should be part of a standard listener that the user should be able to extend but it also could create another listener and never call this method: he would have to manage manually the flags to compute back the new imports then.